notebooks/bedrock/Financial Multilingual Embeddings.ipynb

```diff
@@ -0,0 +1,545 @@
+    "\n",
+    "For this walkthrough you will need to request access to the Cohere Embed Multilingual model."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "494c504f-0886-4b29-8716-0a47e6450647",
+   "metadata": {},
+   "source": [
+    "### Step 1: Install Packages and Import Modules"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bb2388da",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install --upgrade cohere-aws hnswlib translate\n",
+    "# if you upgrade the package, you need to restart the kernel"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "be54602a-944d-43c4-a611-6f8f57285618",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "import cohere_aws\n",
+    "import hnswlib\n",
+    "from translate import Translator\n",
+    "import warnings\n",
+    "import os\n",
+    "import re\n",
+    "import boto3\n",
+    "\n",
+    "warnings.filterwarnings('ignore')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ab9e4080-542b-4aad-94f4-6cf33789c792",
+   "metadata": {},
+   "source": [
+    "### Step 2: Import Documents \n",
+    "\n",
+    "Information about MultiFIN can be found in its Github repo https://github.com/RasmusKaer/MultiFin. The repo contains a link to [Dropbox](https://www.dropbox.com/sh/v0fvtn5gnf4ij6f/AABPSV4NzHIRa8GwUfQ7kA0aa?dl=0) where the data can be downloaded.\n",
+    "\n",
+    "Download the file _MultiFinDataset2023.zip_, unzip the file and navigate into the folder _MultiFinDataset_EACL_ and then _0-all-languages-lowlevel_. It contains a _train.json_ file. Upload that file to your Jupyter SageMaker Notebook."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "94c04e40-e8ca-4273-b4df-d5b8e01990d1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Read the data\n",
+    "df = pd.read_json(\"./train.json\", lines=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7197a0b7-f01e-46bd-ad2d-49395d119cd2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Inspect dataset\n",
+    "df.head(5)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "327177dd-bc70-4259-b3d4-4b7a70eedd1e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# check language distribution\n",
+    "df['lang'].value_counts()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0facfeb0-92e9-4709-b69f-1f2f60db7c43",
+   "metadata": {},
+   "source": [
+    "### Step 3. Select List of Documents to Query"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "fcbafe7d-db5b-4f3d-8234-0ec577761cdd",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Ensure there is no duplicated text in the headers\n",
+    "def remove_duplicates(text):\n",
+    "    return re.sub(r'((\\b\\w+\\b.{1,2}\\w+\\b)+).+\\1', r'\\1', text, flags=re.I)\n",
```

Comment:
this is giving me a headache :)
what exactly is going on; are you sure the regex is the best way to accomplish it; and can you add some detail to the comment explaining why there can't be duplicates?

Reply:
Sorry 😬  I'm picking the longest articles, but some are long just because there are repeated sequences over and over. The regex expressions cleans that up. Perhaps I could saw an example first to show more clearly what we're cleaning.

Reply:
Example of cleaning added in commit 67b5f37ca71a4fbfd4a87a9d1671d3319b7b341a

Reply:
thank you!